app/packages/core/src/components/Grid/useEscape.ts

```diff
@@ -18,7 +18,17 @@ const useEscape = () => {
           }
 
           const modal = await snapshot.getPromise(fos.modalSelector);
-          modal === null && reset(fos.selectedSamples);
+          const selectedSampleIds = await snapshot.getPromise(
+            fos.selectedSamples
+          );
+          // TODO: modal is always `null` here right after a modal closes, so this isn't the condition we want
+          if (modal === null && selectedSampleIds.size > 0) {
```

Comment:
I'm not sure I understand. This seems to work. I assume only if propagation is stopped in looker?

Reply:
I think `modal === null` was put here to try to accomplish what this PR is _actually_ accomplishing. Unfortunately, `modal` is always `null` here, so we do need this PR.